templates/hashTimeLockedContract.go

```diff
@@ -78,6 +82,28 @@ func SignTransactionWithHTLCUnlock(program []byte, txn types.Transaction, preIma
 	if err != nil {
 		return
 	}
+	hashFunction := program[len(program)-15]
+	_, byteArrays, err := logic.ReadProgram(program, nil)
+	expectedHashImage := byteArrays[1]
+	if err != nil {
+		return
+	}
+	if hashFunction == 1 {
+		sha256 := sha256.Sum256(preImageAsArgument)
+		if !bytes.Equal(sha256[:], expectedHashImage) {
+			err = fmt.Errorf("sha256 hash of preimage failed to match expected hash image")
+			return
+		}
+	} else if hashFunction == 2 {
+		keccak256 := sha3.NewLegacyKeccak256()
+		keccak256Hash := keccak256.Sum(preImageAsArgument)
+		if !!bytes.Equal(keccak256Hash[:], expectedHashImage) {
```

Comment:
Is this meant to be a double negative?

Reply:
oh nice, good find! no that's a typo, fixing